pkg/roachprod/vm/azure/azure.go

```diff
@@ -114,6 +117,16 @@ func (p *Provider) GetHostErrorVMs(
 	return nil, nil
 }
 
+type TokenCredential struct {
+	token string
+}
+
+func (t *TokenCredential) GetToken(ctx context.Context, options policy.TokenRequestOptions) (azcore.AccessToken, error) {
```

Comment:
Not sure if i'm missing something obvious as to why azure makes you implement this yourself?? Or why we even explicitly need to pass a token in the first place when the other clients just accept the subID??

Reply:
🤷